kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/streaming/streamExecutionData/SubscribeExecutionDataExample.kt

```
@@ -0,0 +1,70 @@
+                dataChannel.cancel()
+            }
+        }
+
+        val errorJob = scope.launch {
+            try {
+                for (error in errorChannel) {
+                    if (!isActive) break // Respond to cancellation
+                    println("~~~ ERROR: ${error.message} ~~~")
+                    yield() // Cooperative cancellation
+                }
+            } catch (e: CancellationException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)